### PR TITLE
ZEN-10022: Add Multiselect component to the library

### PR DESCRIPTION
**Pull request by leferreyra**
[JIRA Ticket](https://reciprocitylabs.atlassian.net/browse/ZEN-10022)

Copied over the ZenMultiselect component from ZenGRC. There are some visual differences due to the CSS environment. We'll need to figure out what CSS is causing these differences.

#### This is what it looks like when imported in ZenGRC
Screen Shot 2019-08-20 at 11 06 51

#### This is what it looks like in ZenLib's isolated environment (with customized bootstrap)
Screen Shot 2019-08-20 at 11 07 08

This is probably due to some upstream styles in the element hierarchy and the use of relative measurements in the Multiselect styles.

I think components should look the same everywhere so I will probably change all these rems and ems to px at some point.

giphy

### CODE REVIEW

**Comment by leferreyra** [comment]
@mzgajner I have no idea really what's causing this. The thing is also the SASS calculations on relative measurements that are converted to px at build time. For example.

```
$bootstrap-margin: 1rem + 1em + 2px;
...
$multiselect-margin: $bootstrap-margin / 2; // This is converted to some px number at build time.
```
I have no idea if this is the cause, though. I'll have to look into it.

**Comment by vamaq** [comment]
Late LGTM

**Comment by sagaban** [comment]
I don't like the idea of not using rem: if the user want to zoom or if we decide to go responsive, we'll have a lot of troubles. However, I do not like em at all. Maybe another thing we could do is redefine the rem value to 10px, not 16px. A 10 factor pixels is easier to work with: https://medium.com/code-better/css-units-for-font-size-px-em-rem-79f7e592bb97

**Comment by leferreyra** [comment]
@sagaban rems are fine, but when you do arithmetic operations with sass they get converted to px at build time, that's not so cool.

File: packages/multiselect/Multiselect.spec.js [hunk 1]
```diff
@@ -178,4 +178,20 @@ describe("Multiselect.vue", () => {
...
+  describe("renders validation classes", () => {
+    it("renders classes depending on the valid prop", () => {
+      const wrapper = shallowMount(Multiselect);
+      expect(wrapper.attributes("class")).toEqual("");
+
+      wrapper.setProps({ valid: true });
+      expect(wrapper.attributes("class")).toEqual("is-valid");
+
+      wrapper.setProps({ valid: true });
+      expect(wrapper.attributes("class")).toEqual("is-valid");
```
Comment (mplesec) [inline packages/multiselect/Multiselect.spec.js]: This is duplication.

**Comment by mplesec** [comment]
@sagaban Why is using em a problem?

**Pull request merged**